include/picongpu/simulation/stage/AtomicPhysics.hpp

```diff
@@ -570,8 +571,20 @@ namespace picongpu::simulation::stage
     {
         using SpeciesRepresentingAtomicPhysicsIons = particles::atomicPhysics::traits::
             FilterByParticleType_t<VectorAllSpecies, picongpu::particles::atomicPhysics::Tags::Ion>;
-        auto static constexpr numberAtomicPhysicsSpecies = pmacc::mp_size<SpeciesRepresentingAtomicPhysicsIons>::value;
-        static bool constexpr atomicPhysicsActive = (numberAtomicPhysicsSpecies >= 1);
+        auto static constexpr numberAtomicPhysicsIonSpecies
+            = pmacc::mp_size<SpeciesRepresentingAtomicPhysicsIons>::value;
+
+        static bool constexpr atomicPhysicsActive = (numberAtomicPhysicsIonSpecies >= 1);
+
+        // check at least one electron species defined if atomicPhyiscs is active
+        using SpeciesRepresentingAtomicPhysicsElectrons = particles::atomicPhysics::traits::
+            FilterByParticleType_t<VectorAllSpecies, picongpu::particles::atomicPhysics::Tags::Electron>;
+        auto static constexpr numberAtomicPhysicsElectronSpecies
+            = pmacc::mp_size<SpeciesRepresentingAtomicPhysicsElectrons>::value;
+
+        PMACC_CASSERT_MSG(
+            at_least_one_species_marked_as_atomic_physics_electron_species_required,
+            (numberAtomicPhysicsIonSpecies == 0) || (numberAtomicPhysicsElectronSpecies >= 1));
```

Comment:
```suggestion
        // check at least one electron species defined if atomicPhyiscs is active
        using SpeciesRepresentingAtomicPhysicsElectrons = particles::atomicPhysics::traits::
            FilterByParticleType_t<VectorAllSpecies, picongpu::particles::atomicPhysics::Tags::Electron>;
        auto static constexpr numberAtomicPhysicsElectronSpecies
            = pmacc::mp_size<SpeciesRepresentingAtomicPhysicsElectrons>::value;

        static bool constexpr atomicPhysicsActive = (numberAtomicPhysicsIonSpecies > 0 && numberAtomicPhysicsElectronSpecies > 0);

        PMACC_CASSERT_MSG(
        at_least_one_atomic_physics_ion_and_one_atomic_physics_electron_species_required_for_atomic_physics,
            (numberAtomicPhysicsIonSpecies > 0) && (numberAtomicPhysicsElectronSpecies > 0));
```
I think it might be a good idea to check both ions and electrons for `atomicPhysicsActive` and also raise the `CASSERT` with the same check

Reply:
discussion offline, checking for existence of both atomic physics ions and electrons for `atomicPhysicsActive` is a good idea, but we may not include the check for atomic physics ions in the compile time assert since, this would always cause a compile error if no atomic physics ions are configured in the simulation, making a simulation without atomicPhysics impossible.

Reply:
implemented